packages/blade/src/components/Input/BaseInput/BaseInputVisuals.tsx

```diff
@@ -314,24 +317,30 @@ export const BaseInputVisuals = ({
     );
   }
 
+  if (hasTrailingVisuals && hasTrailingInteractionElement) {
+    return (
+      <BaseBox {...getVisualContainerStyles()} alignSelf="stretch">
+        <BaseBox
+          paddingRight={getInteractionElementStyles({
+            hasTrailingIcon,
+            hasTrailingInteractionElement,
+            hasSuffix,
+            hasTrailingButton,
+          })}
+          display="flex"
+          alignItems="stretch"
+          alignSelf="stretch"
+          {...(!isReactNative() && { onClick: onTrailingInteractionElementClick })}
+        >
+          {trailingInteractionElement}
+        </BaseBox>
+      </BaseBox>
+    );
+  }
+
   if (hasTrailingVisuals) {
     return (
       <BaseBox alignSelf="stretch" alignItems="stretch" {...getVisualContainerStyles()}>
-        {hasTrailingInteractionElement ? (
```

Comment:
see if you can keep this code and make minimal changes with conditions. baseinput has too many gotchas because its used in all inputs 😢